dev/AppNotifications/AppNotificationBuilder/AppNotificationBuilder.idl

```diff
@@ -1,15 +1,28 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 import "..\AppNotifications\AppNotifications.idl";
 
 namespace Microsoft.Windows.AppNotifications.Builder
 {
-    [contractversion(1)]
+    [contractversion(2)]
     apicontract AppNotificationBuilderContract {}
 
+    [contract(AppNotificationBuilderContract, 2)]
+    runtimeclass AppNotificationHeader
+    {
+        AppNotificationHeader();
+        AppNotificationHeader(String id, String title);
+
+        String Id;
+        String Title;
+        Windows.Foundation.Collections.IMap<String, String> Arguments;
+
+        AppNotificationHeader AddArgument(String key, String value);
```

Comment:
Why an `AddArgument` method when `Arguments` is exposed? Why not `header.Arguments.Add(x,y,)`?

If the arguments here are meant to be read-only then that would be `IMapView` instead of `IMap` (though one would then ask Where's the RemoveArgument()?...)

Reply:
The fluent setters work in any languages and let developers specify complex AppNotification using the dot notation. The property accessors are useful in langugages that support object initialization syntax (C# does but C++ does not, for example) and offer an alternate syntax to the dot notation afforded by the fluent setter by letting developers configure objects at construction time.

This is the general pattern already established by the AppNotificationBuilder.